### PR TITLE
feat: Argument to select moved block in `moveBlocksUp/Down` (BLO-907)

### LINKED ISSUE #1414: Export editor.moveBlocksUp() with a optional parameter blockId, or export another function replace.
**Is your feature request related to a problem? Please describe.**
I want to add `Move Up` and `Move Down` item in `DragHandleMenu`, but `editor.moveBlocksUp()` can only move up the selected blocks or current cursor, and the invoked function in `moveBlocksUp(editor)` are not exported. So can you export `editor.moveBlocksUp()` and `editor.moveBlocksDown()` with a optional parameter blockId, or export another function replace?

**Describe the solution you'd like**
Export  `editor.moveBlocksUp()` and `editor.moveBlocksDown()` with a optional parameter blockId, or export another function replace.

**Describe alternatives you've considered**

**Additional context**

**Bonus**
[ ] I'm a [sponsor](https://github.com/sponsors/YousefED) and would appreciate if you could look into this sooner than later 💖

### PR DESCRIPTION
**Pull request by matthewlipski**
# Summary

This PR adds a `blockIdentifier` optional argument to `moveBlocksUp` and `moveBlocksDown`. If specified, the provided block is moved up/down. Nothing happens if the block could not be found. In not provided, the selected blocks are moved, i.e. the same behaviour as before is used.

This PR also fixes some edge cases when the selection starts in, ends in, or spans columns.

Closes #1414

## Rationale

This makes the functions more useful for consumers.

## Changes

- Added `moveBlocks` helper function, which also includes logic to handle columns.
- Added optional `blockIdentifier: BlockIdentifier` argument to `moveBlocksUp` and `moveBlocksDown`.
- Updated docs.

## Impact

N/A

## Testing

Added unit tests.

## Screenshots/Video

N/A

## Checklist

- [X] Code follows the project's coding standards.
- [X] Unit tests covering the new feature have been added.
- [X] All existing tests pass.
- [X] The documentation has been updated to reflect the new feature

## Additional Notes

N/A

## Summary by CodeRabbit

* **New Features**
  * Block move actions now accept an optional explicit block identifier; specifying a block moves that block while preserving the current selection, otherwise the selection is moved.

* **Documentation**
  * API docs updated with overload signatures and examples for both selection-based and explicit-block moves.

* **Tests**
  * Added tests for explicit-block moves, selection stability, no-op edge cases, nested/multi-column scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/TypeCellOS/BlockNote/pull/2723)

### CODE REVIEW

File: packages/core/src/api/blockManipulation/commands/moveBlocks/moveBlocks.ts [hunk 1]
```diff
@@ -148,6 +148,27 @@ function flattenColumns(
...
+  editor: BlockNoteEditor<any, any, any>,
+  blocks: Block<any, any, any>[],
+  referenceBlock: BlockIdentifier,
+  placement: "before" | "after",
+) {
+  editor.transact(() => {
+    editor.removeBlocks(blocks);
+    editor.insertBlocks(flattenColumns(blocks), referenceBlock, placement);
+  });
+}
```
[the remark this answers is not in the document]
Comment (nperez0111) [reply packages/core/src/api/blockManipulation/commands/moveBlocks/moveBlocks.ts:170]: This is actually probably valid @matthewlipski

**Review by nperez0111 — APPROVED** [review]
